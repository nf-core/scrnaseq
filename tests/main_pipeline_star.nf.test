nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("test-dataset_star_aligner") {

        when {
            // the rest is taken from shared config
            params {
                aligner    = 'star'
                outdir     = "${outputDir}/results_star"
            }
        }

        then {

            assertAll(

                //
                // General assertions
                //

                // Did it finish successfully?
                {assert workflow.success},

                // How many tasks were executed?
                {assert workflow.trace.tasks().size() == 19},

                // How many results were produced?
                {assert path("${outputDir}/results_star").list().size() == 4},
                {assert path("${outputDir}/results_star/star").list().size() == 3},
                {assert path("${outputDir}/results_star/star/mtx_conversions").list().size() == 8},
                {assert path("${outputDir}/results_star/fastqc").list().size() == 12},
                {assert path("${outputDir}/results_star/multiqc").list().size() == 3},

                //
                // Check if files were produced
                //
                {assert new File( "${outputDir}/results_star/star/mtx_conversions/Sample_X/Sample_X_raw_matrix.h5ad" ).exists()},
                {assert new File( "${outputDir}/results_star/star/mtx_conversions/Sample_Y/Sample_Y_raw_matrix.h5ad" ).exists()},
                {assert new File( "${outputDir}/results_star/star/mtx_conversions/Sample_X/Sample_X_filtered_matrix.h5ad" ).exists()},
                {assert new File( "${outputDir}/results_star/star/mtx_conversions/Sample_Y/Sample_Y_filtered_matrix.h5ad" ).exists()},

                //
                // Check if files are the same
                //
                {assert snapshot(
                    path( "${outputDir}/results_star/star/Sample_X/Sample_X.SJ.out.tab"                             ),
                    path( "${outputDir}/results_star/star/Sample_X/Sample_X.Solo.out/Barcodes.stats"                ),
                    path( "${outputDir}/results_star/star/Sample_X/Sample_X.Solo.out/Gene/filtered/matrix.mtx.gz"   ),
                    path( "${outputDir}/results_star/star/Sample_X/Sample_X.Solo.out/Gene/filtered/features.tsv.gz" ),
                    path( "${outputDir}/results_star/star/Sample_X/Sample_X.Solo.out/Gene/filtered/barcodes.tsv.gz" ),
                    path( "${outputDir}/results_star/star/Sample_Y/Sample_Y.SJ.out.tab"                             ),
                    path( "${outputDir}/results_star/star/Sample_Y/Sample_Y.Solo.out/Barcodes.stats"                ),
                    path( "${outputDir}/results_star/star/Sample_Y/Sample_Y.Solo.out/Gene/filtered/matrix.mtx.gz"   ),
                    path( "${outputDir}/results_star/star/Sample_Y/Sample_Y.Solo.out/Gene/filtered/features.tsv.gz" ),
                    path( "${outputDir}/results_star/star/Sample_Y/Sample_Y.Solo.out/Gene/filtered/barcodes.tsv.gz" ),
                    path( "${outputDir}/results_star/star/mtx_conversions/Sample_X/Sample_X_raw_matrix.seurat.rds"  ),
                    path( "${outputDir}/results_star/star/mtx_conversions/Sample_X/Sample_X_raw_matrix.sce.rds"     ),
                    path( "${outputDir}/results_star/star/mtx_conversions/Sample_Y/Sample_Y_raw_matrix.seurat.rds"  ),
                    path( "${outputDir}/results_star/star/mtx_conversions/Sample_Y/Sample_Y_raw_matrix.sce.rds"     ),
                    path( "${outputDir}/results_star/star/mtx_conversions/Sample_X/Sample_X_filtered_matrix.seurat.rds" ),
                    path( "${outputDir}/results_star/star/mtx_conversions/Sample_X/Sample_X_filtered_matrix.sce.rds"    ),
                    path( "${outputDir}/results_star/star/mtx_conversions/Sample_Y/Sample_Y_filtered_matrix.seurat.rds" ),
                    path( "${outputDir}/results_star/star/mtx_conversions/Sample_Y/Sample_Y_filtered_matrix.sce.rds"    )
                ).match()}

            ) // end of assertAll()

        }
    }

}
