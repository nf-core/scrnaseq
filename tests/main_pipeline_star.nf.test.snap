{
    "test-dataset_star_aligner": {
        "content": [
            "Sample_X.SJ.out.tab:md5,d2d7f0abe38029012571bdf6622fc6eb",
            "Barcodes.stats:md5,7f99dc8aa5e074fbe5779ea7712c0886",
            "matrix.mtx.gz:md5,6a923393343aa1a69b0cf1bd998c9285",
            "features.tsv.gz:md5,99e453cb1443a3e43e99405184e51a5e",
            "barcodes.tsv.gz:md5,9a7dacaa1779ea43c1507a947fe6992a",
            "Sample_Y.SJ.out.tab:md5,98bd31104a860cf80119dc30d938d163",
            "Barcodes.stats:md5,2dbf1ae426c1afd97903ee001f0db5ce",
            "matrix.mtx.gz:md5,0ae080bd0002e350531a5816e159345e",
            "features.tsv.gz:md5,99e453cb1443a3e43e99405184e51a5e",
            "barcodes.tsv.gz:md5,9b695b0b91bcb146ec9c4688ca10a690",
            "Sample_X_raw_matrix.seurat.rds:md5,34fd07749de1f1ee49e885782cd7c6c5",
            "Sample_X_raw_matrix.sce.rds:md5,ef88d8a6bb48eb8f555a0c255d9b03bb",
            "Sample_Y_raw_matrix.seurat.rds:md5,5b8361f36fec65107e7652751c80fdb5",
            "Sample_Y_raw_matrix.sce.rds:md5,d9d6c934d2193681ab58a88d437bf138",
            "Sample_X_filtered_matrix.seurat.rds:md5,05780c38d8deb9d06384da2c12cbdefd",
            "Sample_X_filtered_matrix.sce.rds:md5,41f1959bd1c577305915fdfcb797bf0b",
            "Sample_Y_filtered_matrix.seurat.rds:md5,d27d6360cf500bf4438f650b8796f8ce",
            "Sample_Y_filtered_matrix.sce.rds:md5,2c63e5563a037d2834a6eef107b34f67"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-11-29T15:32:12.524479228"
    }
}
